src/app/dynamic/enterprise/metering/component.ts

```diff
@@ -70,6 +70,11 @@ export class MeteringComponent implements OnInit {
       });
   }
 
+  ngOnDestroy(): void {
```

Comment:
This is unused until class implements OnDestroy.

Reply:
Are you sure about it? I just checked and putting console.log in the method printed message in the console after I switched the component. What you say makes sense and I'll add this interface to the class, just wanted to point out that it seems to work even without explicitly adding the interface to class definition.

Reply:
@pkprzekwas 

Yes! interfaces are optional Angular lifecycle hooks are called even you don't implement it. This is helpful for devs. TypeScript compiler warns you if you don't implement the Lifecycle methods correctly but it is still executed incase of 
- pipes, services, directive and components

[Docs](https://v8.angular.io/guide/lifecycle-hooks#interfaces-are-optional-technically)

Reply:
I was not aware of it but anyways it would be better to have it :)